File: FirClient/Assets/Scripts/Lua/Data/Tables/QualityTable.lua
local QualityTable = class("QualityTable")

function QualityTable:Initialize()
    self.Items = {
        [1] = {id = 1, name = '灰色品质框', icon = 'gray'},
        [2] = {id = 2, name = '蓝色品质框', icon = 'blue'},
        [3] = {id = 3, name = '绿色品质框', icon = 'green'},
        [4] = {id = 4, name = '紫色品质框', icon = 'purple'},
        [5] = {id = 5, name = '橙色品质框', icon = 'orange'},
        [6] = {id = 6, name = '金色品质框', icon = 'yellow'}
	}
end

function QualityTable:GetItems()
    return self.Items
end

function QualityTable:GetItemByKey(key)
    return self.Items[key]
end

return QualityTable

File: FirServer/GameLibs/Data/Tables/QualityTable.cs
using System;
using System.Collections.Generic;
using UnityEngine;

namespace FirCommon.Data
{
	public class QualityTable
	{
		public string name;		
		private List<QualityTableItem> items = new List<QualityTableItem>();
		
		private Dictionary<int, QualityTableItem> dics = null;

		public List<QualityTableItem> Items
		{
			get {
				return items;
			}
		}

		public void Initialize()
		{
			dics = new Dictionary<int, QualityTableItem>();
			foreach (QualityTableItem item in items)
			{
				dics.Add(item.id, item);
			}
		}

		public List<QualityTableItem> GetItems()
		{
			return items;
		}

		public void AddItem(QualityTableItem item)
		{
			items.Add(item);
		}

		public QualityTableItem GetItemByKey(int key)
		{
			QualityTableItem item = null;
			if (dics.ContainsKey(key))
			{
				dics.TryGetValue(key, out item);
			}
			return item;
		}
	}

	public class QualityTableItem
	{
    	public int id;
    	public string name;
    	public string icon;
	}
}